dex/src/main/scala/com/wavesplatform/dex/tool/WaitOffsetTool.scala

```diff
@@ -0,0 +1,58 @@
+package com.wavesplatform.dex.tool
+
+import akka.actor.ActorSystem
+import com.wavesplatform.dex.domain.utils.ScorexLogging
+import com.wavesplatform.dex.queue.ValidatedCommandWithMeta.Offset
+import com.wavesplatform.dex.settings.MatcherSettings
+
+import java.util.concurrent.TimeoutException
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.concurrent.duration.Deadline
+import scala.util.{Failure, Success}
+
+object WaitOffsetTool extends ScorexLogging {
+
+  def waitOffsetReached(
+    getLastOffset: Deadline => Future[Offset],
+    getLastProcessedOffset: => Offset,
+    startingLastOffset: Offset,
+    deadline: Deadline,
+    settings: MatcherSettings
+  )(implicit actorSystem: ActorSystem, ex: ExecutionContext): Future[Unit] = {
+    val commandsPerSecond = settings.waitingQueueSettings.commandsPerSecond
+    val interval = settings.waitingQueueSettings.checkInterval
+    val maxWaitingTime = settings.waitingQueueSettings.maxWaitingTime
+
+    def canProcessNewCommands(lastOffset: Offset, currentOffset: Offset): Boolean = {
+      val leftTime = (lastOffset - currentOffset) / commandsPerSecond
+      log.trace(s"Left time before matcher starts: $leftTime")
+      leftTime <= maxWaitingTime.toSeconds
```

Comment:
Actually, we don't know `commandsPerSecond` and it would be nice to calculate it. To do this, we need two pairs of:
1. The time (`System.nanoTime`) of a measurement
2. The offset at this time

At first time we could just compare `currentOffset` and `lastOffset`, because there was no a previous measurement.

For subsequent checks we could calculate the `commandsPerSecond: Double = (currentOffset - previousOffset).toDouble / (currentTs - previousTs)`